core/services/ocr2/plugins/ocr2keeper/evm/registry.go

```diff
@@ -83,6 +128,20 @@ func NewEVMRegistryServiceV2_0(addr common.Address, client evm.Chain, mc *models
 	return r, nil
 }
 
+func setupCaches(defaultUpkeepExpiration, defaultCooldownExpiration, defaultApiErrExpiration, cleanupInterval time.Duration) (*cache.Cache, *cache.Cache, *cache.Cache) {
+	// cache that stores UpkeepInfo for callback during MercuryLookup
+	upkeepInfoCache := cache.New(defaultUpkeepExpiration, cleanupInterval)
```

Comment:
cc @amirylm i think this has some intersection with the upkeep life cycle you are working on

Reply:
It does, I planned to have some sort of cache for upkeeps (`EvmRegistry.active` today)  that can be used instead of additional one.
see https://github.com/smartcontractkit/chainlink/pull/9141/files#diff-03fc70baf7d43364a13a6a5439004d3db25ad6c20fbbfb253dd449c671a6f215R121